src/collections/vector.ts

```diff
@@ -15,15 +14,13 @@ function indexToKey(prefix: Bytes, index: number): Bytes {
 
 /// An iterable implementation of vector that stores its content on the trie.
 /// Uses the following map: index -> element
-export class Vector<E> {
+export class Vector {
```

Comment:
Curious why the removal of the generic? Won't using `unknown` be limiting for developers? (Asking just to learn/understand, ignore if you'd like)

Reply:
Version with generics appeared to be buggy and resource hungry. Deserialization of the class object is not a trivial task, it is still an open question of how much magic we want to have, and how much should be done by the user, so he is in control of all the computations happening on the chain.
The current design is not final, this is something stable we were able to ship right now.